Assets/Scripts/Tools/ExtendedRandom.cs

```diff
@@ -0,0 +1,144 @@
+using UnityEngine;
+
+namespace ExtendedRandom
+{
+    /// <summary>
+    /// Class for random with various random functions
+    /// </summary>
+    public class RandomX
+    {
+        readonly private System.Random random;
+
+        public RandomX()
+        {
+            random = new System.Random();
+        }
+
+        public RandomX(int seed)
+        {
+            random = new System.Random(seed);
+        }
+
+        /// <summary>
+        /// Returns a random point inside or on a circle with radius 1.0
+        /// </summary>
+        public Vector2 InsideUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi]
+            float radius = Value(); // Generate radius between [0, 1]
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: radius * Mathf.Cos(angle),
+                y: radius * Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// Returns a random point on the edge of a circle with radius 1.0
+        /// </summary>
+        public Vector2 OnUnitCircle()
+        {
+            float angle = Value(0, 2) * Mathf.PI; // Generate angle between [0, 2*Pi]
+
+            // Convert to from polar coordinates to regular coordinates
+            return new Vector2(
+                x: Mathf.Cos(angle),
+                y: Mathf.Sin(angle));
+        }
+
+        /// <summary>
+        /// Returns a random point inside or on a sphere with radius 1.0
+        /// </summary>
+        public Vector3 InsideUnitSphere()
+        {
+            float phi = Value(0, 2) * Mathf.PI; // Generates angle between [0, 2*Pi]
+            float theta = Value() * Mathf.PI; // Generates angle between [0, Pi]
+            float radius = Value(); // Generates radius between [0, 1]
+
+            // Convert from spherical coordinates to regular coordinates
+            return new Vector3(
+                x: radius * Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: radius * Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: radius * Mathf.Cos(theta));
+        }
+
+        /// <summary>
+        /// Returns a random point on the surface of a sphere with radius 1.0
+        /// </summary>
+        public Vector3 OnUnitSphere()
+        {
+            float phi = Value(0, 2) * Mathf.PI; // Generates angle between [0, 2*Pi]
+            float theta = Value() * Mathf.PI; // Generates angle between [0, Pi]
+
+            // Convert from spherical coordinates to regular coordinates
+            return new Vector3(
+                x: Mathf.Sin(theta) * Mathf.Cos(phi),
+                y: Mathf.Sin(theta) * Mathf.Sin(phi),
+                z: Mathf.Cos(theta));
+        }
```

Comment:
This makes points close to the center more likely. I guess we just ignore that? Other option would be randomising xyz within [-range, range] and checking if inside sphere.

Reply:
fixed